### PR TITLE
added typescript type definitions

### LINKED ISSUE #38: Typescript Support

### PR DESCRIPTION
**Pull request by Kylof61034**
closes #38

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        playlistId: string;
+        playlistThumbnail: string;
+        playlistUrl: string;
+        title: string;
+        type: "playlist";
+        videoCount: number;
+    }
+
+    class YoutubeGrabber {
+        static getChannelInfo(channelId: string): Promise<ChannelInfo>;
```
Comment (PikachuEXE) [inline index.d.ts:81]: Seems missing optional argument `channelIdType`

Same for many methods below

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        static getChannelVideos(channelId: string, sortBy?: "newest" | "oldest" | "popular"): Promise<ChannelInfoResponse<Video>>;
```
Comment (PikachuEXE) [inline index.d.ts:83]: Seems missing optional argument `channelIdType`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        static getChannelVideosMore(continuation: string): Promise<ChannelInfoResponse<Video>>;
+
+        static getChannelPlaylistInfo(channelId: string, sortBy?: "last" | "oldest" | "newest"): Promise<ChannelInfoResponse<Playlist>>;
```
Comment (PikachuEXE) [inline index.d.ts:87]: Seems missing optional argument `channelIdType`

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+        static getChannelPlaylistsMore(continuation: string): Promise<ChannelInfoResponse<Playlist>>;
+
+        static searchChannel(channelId: string, query: string): Promise<ChannelInfoResponse<Video>>;
+
+        static searchChannelMore(continuation: string): Promise<ChannelInfoResponse<Video>>;
+    }
```
Comment (PikachuEXE) [inline index.d.ts:94]: Can't find `getChannelCommunityPosts` or `getChannelCommunityPostsMore`

**Review by PikachuEXE — CHANGES_REQUESTED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Maybe also take this one over. Author hasn't been very active on GH.

**Review by PikachuEXE — DISMISSED** [review]
I cannot be sure all content in `index.d.ts` matches the doc (Too long to review all)
But since it can always be fixed later I will approve

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Review by PikachuEXE — DISMISSED** [review]

**Comment by Slayug** [comment]
Are the last reviewers are aware of this ?  
Do you think it possible to bypass all requirements ?

**Comment by PikachuEXE** [comment]
The dev team might be on vacation in Dec.
Thanks for checking the progress I think they will be back in Jan.
Is there any urgent need for this to be merged?

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        CustomURL
+    }
+
+    /**
+     * A abstract ChannelInfoResponse containing a continuation string if there are more responses which can be loaded
+     * and an array of items of the type T
+     */
+    interface ChannelInfoResponseContinuation<T> {
+        items: T[];
+        continuation: string | null; // Will return null if no more results can be found.  Used with getChannelPlaylistsMore()
```
Comment (Slayug) [inline index.d.ts:18]:
```suggestion
        /**
        * Will be null if no more results can be found.  Used with getChannelPlaylistsMore()
        */
        continuation: string | null;
```
It's better to use multi-line comment to have the code helper on hover

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    /**
+     * ChannelInfo type returned by getChannelVideos and getChannelInfoMore
+     */
+    interface ChannelInfo {
+        author: string;
+        authorId: string;
+        authorUrl: string;
+        authorBanners: Image[] | null; // Will return null if none exist
```
Comment (Slayug) [inline index.d.ts:79]:
```suggestion
        /**
        * Is null if none exist
        **/
        authorBanners: Image[] | null;
```

File: index.d.ts [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        authorThumbnails: Image[] | null; // Will return null if none exist
```
Comment (Slayug) [inline index.d.ts:80]:
```suggestion
        /**
        * Is null if none exist
        **/
        authorThumbnails: Image[] | null;
```

**Review by Slayug — COMMENTED** [review]
better usage of comment fields

**Comment by Slayug** [comment]
@PikachuEXE thought dev team was inactive because there is no activity since october

**Comment by ChunkyProgrammer** [comment]
> @PikachuEXE thought dev team was inactive because there is no activity since october

I think the team has just been extra busy recently with work, school and personal stuff

**Comment by PikachuEXE** [comment]
I don't check this repo at all unless got mentioned / review requested
I normally only check the main repo (if I remember to)

There are also a lot of issues/PR on main repo which also drains away our attention
So if there is a good reason to raise the urgency of this PR, feel free to state / discuss here (I think I am active for discussion at least)
But let's also adjust the expectation of dev team availability

Most of us don't get paid (or not much) to maintain open source projects, meaning it's charity work
And I got multiple (not just this) open source projects to handle and a day job to take care
I can respond within a week or even a day mostly, but no guarantee
I would expect the other can be busier than me

**Review by PikachuEXE — DISMISSED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Comment by ChunkyProgrammer** [comment]
> Why would this PR also update `app/youtube-grabber.js`?

I'm not sure, seems to be some merge issue

**Comment by PikachuEXE** [comment]
CI failed due to service returning 503, not related

**Review by PikachuEXE — APPROVED** [review]

**Review by absidue — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Pull request merged**